cdk/transactionAnalyzer/src/main/java/software/amazon/event/kafkaconnector/TransactionAnalyzer.java

```diff
@@ -0,0 +1,164 @@
+/*
+ *
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ *
+ */
+
+package software.amazon.event.kafkaconnector;
+
+import com.amazonaws.services.schemaregistry.kafkastreams.AWSKafkaAvroSerDe;
+import com.amazonaws.services.schemaregistry.utils.AWSSchemaRegistryConstants;
+import com.amazonaws.services.schemaregistry.utils.AvroRecordType;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.common.config.TopicConfig;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.kstream.KStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * This class reads records from an input topic (events), filters events which have a total of over 180 and sends those
+ * to an output topic (notifications) using Kafka Streams
+ */
+public class TransactionAnalyzer {
+    private static final Logger log = LoggerFactory.getLogger(TransactionAnalyzer.class);
+    private static final Properties props = new Properties();
+    private static final Schema.Parser parser = new Schema.Parser();
+    private static final CountDownLatch latch = new CountDownLatch(1);
+
+    public static void main(String[] args) {
+
+        var sourceTopic = System.getenv().getOrDefault("TOPIC_NAME", "events");
+        var notificationsTopic = System.getenv().getOrDefault("NOTIFICATIONS_TOPIC_NAME", "notifications");
+        var replicationFactor = System.getenv().getOrDefault("REPLICATION_FACTOR", "3");
+        var partitionCount = System.getenv().getOrDefault("PARTITION_COUNT", "10");
+        var notificationSchemaFile = System.getenv().getOrDefault("NOTIFICATION_SCHEMA_FILE", "notification.avsc");
+
+        var notificationSchema = createSchema(notificationSchemaFile);
+        var properties = setConfiguration();
+
+        var topicCreator = new TransactionAnalyzerTopicCreator(properties);
+        topicCreator.createTopic(notificationsTopic, Integer.valueOf(partitionCount), Short.valueOf(replicationFactor));
+        topicCreator.createTopic(sourceTopic, Integer.valueOf(partitionCount), Short.valueOf(replicationFactor));
```

Comment:
Shouldn't `sourceTopic` already exist? Or is this to prevent races between producer/consumer startup?